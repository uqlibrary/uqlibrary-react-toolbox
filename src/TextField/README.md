# TextField

A simple TextField component that lists items for UQ Library applications using ReactJs

## Props
- `standard props from http://www.material-ui.com/#/components/text-field`
- `helpTitle: PropTypes.string`
- `helpText: PropTypes.any`


## Usage

**./App.js**
```jsx
import React from 'react';
...

import {TextField} from 'uqlibrary-react-toolbox';
 
const App = () => (
  <div>
    <TextField 
        name="sampleFieldName" 
        type="text" 
        fullWidth
        floatingLabelText="Sample floating label"/>
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
