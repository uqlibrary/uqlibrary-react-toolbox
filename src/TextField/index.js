export {default as TextField} from '../TextField/components/Textfield';
