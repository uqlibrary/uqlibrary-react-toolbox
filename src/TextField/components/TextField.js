import React from 'react';
import TextField from 'material-ui/TextField';
import PropTypes from 'prop-types';

import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

const TextFieldWrapper = props => {
    const filteredProps = propFilter(props, TextField.propTypes);
    delete filteredProps.className;
    const classProps = props.className ? props.className + ' input-long-hint' : 'input-long-hint';
    return (
        <div style={{position: 'relative', width: '100%'}}>
            <TextField {...filteredProps} className={classProps} />
            {props.help && props.help.text && (
                <HelpIcon {...props.help} />
            )}
        </div>
    );
};

TextFieldWrapper.propTypes = {
    ...TextField.propTypes,
    help: PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.any,
        buttonLabel: PropTypes.string
    })
};

export default TextFieldWrapper;
