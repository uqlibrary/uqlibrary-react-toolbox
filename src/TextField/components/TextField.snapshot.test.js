jest.dontMock('./TextField');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import TextField from './TextField';

import filterProps from '../../helpers/_filterProps';

function setup(props) {
    const consolidatedProps = filterProps(props, TextField.propTypes);
    return shallow(<TextField {...consolidatedProps} />);
}

describe('TextFieldWrapper snapshots tests', () => {
    it('renders TextField component', () => {
        const props =
            {
                name: 'testField',
                type: 'text',
                fullWidth: true,
                floatingLabelText: 'This is a test textfield component'
            };

        const wrapper = setup(props);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
