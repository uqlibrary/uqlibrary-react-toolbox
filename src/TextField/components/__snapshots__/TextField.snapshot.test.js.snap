// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldWrapper snapshots tests renders TextField component 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextField
    className=" mui-long-labels-fix"
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="This is a test textfield component"
    fullWidth={true}
    multiLine={false}
    name="testField"
    rows={1}
    type="text"
    underlineShow={true}
  />
</div>
`;

exports[`TextFieldWrapper snapshots tests renders TextField component appending a class 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextField
    className="requiredField mui-long-labels-fix"
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="This is a test textfield component"
    fullWidth={true}
    multiLine={false}
    name="testField"
    rows={1}
    type="text"
    underlineShow={true}
  />
</div>
`;
